pkg/schemadsl/compiler/translator.go

```diff
@@ -258,7 +285,12 @@ func translateObjectDefinition(tctx *translationContext, defNode *dslNode) (*cor
 		return ns, nil
 	}
 
-	ns := namespace.Namespace(nspath, relationsAndPermissions...)
+	var ns *core.NamespaceDefinition
+	if slices.Contains(tctx.enabledFlags, "deprecation") && slices.Contains(tctx.allowedFlags, "deprecation") && (deprecationForNamespace != &core.Deprecation{}) {
```

Comment:
if deprecation is not enabled, we should make sure to raise an error during parsing